src/algorithms/calorimetry/CalorimeterHitDigi.cc

```diff
@@ -232,6 +232,7 @@ void CalorimeterHitDigi::process(const CalorimeterHitDigi::Input& input,
                                             n_max_photons * m_cfg.capADC),
                      0LL);
     } else if (readoutType == kSipmReadout) {
+      edep *= (1.0 + eResRel);
```

Comment:
Put before this `if` block so that `kSimpleReadout` and `kPoissonPhotonReadout` cases are supported.

Reply:
I don't think it makes sense to handle `kPoissonPhotonReadout` differently. It's not used by any calorimeter. and is mostly for testing questions like "what is our kSipmReadout calorimeter didn't have saturation". If you are concerned that other users may need to smear after saturation (e.g. dominated by electronics), you'd need to add a switch to support their case.